web_src/js/features/repo-home.ts

```diff
@@ -7,141 +7,144 @@ import {fomanticQuery} from '../modules/fomantic/base.ts';
-  });
+    mgrBtn.addEventListener('click', () => {
+      hideElem(viewDiv);
+      showElem(editDiv);
+      topicDropdown.querySelector<HTMLInputElement>('input.search').focus();
+    });
 
-  document.querySelector('#cancel_topic_edit').addEventListener('click', () => {
-    lastErrorToast?.hideToast();
-    hideElem(editDiv);
-    showElem(viewDiv);
-    mgrBtn.focus();
-  });
+    mgrBtn.parentNode.querySelector(`#cancel_topic_edit_${viewType}`).addEventListener('click', () => {
+      lastErrorToast?.hideToast();
+      hideElem(editDiv);
+      showElem(viewDiv);
+      mgrBtn.focus();
+    });
 
-  document.querySelector('#save_topic').addEventListener('click', async (e) => {
-    lastErrorToast?.hideToast();
-    const topics = editDiv.querySelector('input[name=topics]').value;
+    mgrBtn.parentNode.querySelector(`#save_topic_${viewType}`).addEventListener('click', async (e: MouseEvent & {target: HTMLButtonElement}) => {
+      lastErrorToast?.hideToast();
+      const topics = editDiv.querySelector<HTMLInputElement>('input[name=topics]').value;
 
-    const data = new FormData();
-    data.append('topics', topics);
+      const data = new FormData();
+      data.append('topics', topics);
 
-    const response = await POST(e.target.getAttribute('data-link'), {data});
+      const response = await POST(e.target.getAttribute('data-link'), {data});
 
-    if (response.ok) {
-      const responseData = await response.json();
-      if (responseData.status === 'ok') {
-        queryElemChildren(viewDiv, '.repo-topic', (el) => el.remove());
-        if (topics.length) {
-          const topicArray = topics.split(',');
-          topicArray.sort();
-          for (const topic of topicArray) {
-            // it should match the code in repo/home.tmpl
-            const link = document.createElement('a');
-            link.classList.add('repo-topic', 'ui', 'large', 'label');
-            link.href = `${appSubUrl}/explore/repos?q=${encodeURIComponent(topic)}&topic=1`;
-            link.textContent = topic;
-            mgrBtn.parentNode.insertBefore(link, mgrBtn); // insert all new topics before manage button
+      if (response.ok) {
+        const responseData = await response.json();
+        if (responseData.status === 'ok') {
+          queryElemChildren(viewDiv, '.repo-topic', (el) => el.remove());
+          if (topics.length) {
+            const topicArray = topics.split(',');
+            topicArray.sort();
+            for (const topic of topicArray) {
+              // it should match the code in repo/home.tmpl
+              // TODO: sort items in topicDropdown, or items in edit div will have different order to the items in view div
+              const link = document.createElement('a');
+              link.classList.add('repo-topic', 'ui', 'large', 'label');
+              link.href = `${appSubUrl}/explore/repos?q=${encodeURIComponent(topic)}&topic=1`;
+              link.textContent = topic;
+              viewDiv.append(link);
```

Comment:
Since you have 2 buttons, I do not think the logic is right.

With wide view, edit and some items appear, then make the view narrow, then items are out-of-sync?

Reply:
If we should also sync editing status, it is hard for me to implement now. (have no idea about how to implement it now)